src/lib.rs

```diff
@@ -466,7 +466,7 @@ pub type VkInstance = Handle<back::Instance>;
 pub type VkPhysicalDevice = Handle<hal::Adapter<B>>;
 pub type VkDevice = Handle<hal::Gpu<B>>;
 pub type VkCommandPool = Handle<<B as hal::Backend>::CommandPool>;
-pub type VkCommandBuffer = <B as hal::Backend>::CommandBuffer;
+pub type VkCommandBuffer = Handle<<B as hal::Backend>::CommandBuffer>;
```

Comment:
that is precisely what I want to avoid. Having the command buffer in a box means an indirection on each call...
Doesn't Vulkan.h define the size of `CommandBuffer`? What makes you think it has to be a single word?

Reply:
Yes, vulkan.h defines the size. It's a ptr/u64 handle (https://github.com/KhronosGroup/Vulkan-Docs/blob/1.0/src/vulkan/vulkan.h#L53-L62). We have to wrap it as the size of a command buffer is different for each backend. Currently, we allocate a `VkCommandBuffer` on the stack (64bit) and allocate command buffers from the pool by writing a `<B as hal::Backend>::CommandBuffer` to the 64bit stack value and free this 64bit value later by interpreting it as `<B as hal::Backend>::CommandBuffer`, which requires more space causing a stack corruption.